### PR TITLE
Implementation of non-local method for fully integrated thickshells Isolid=14

### PR DESCRIPTION
**Pull request by vdavaze**
#### Description of the feature or the bug

Non-local method was not compatible with thickshells Isolid=14. 

#### Description of the changes

Add the compatibility, + fix minor bugs and update error messages.

### CODE REVIEW

File: engine/source/elements/thickshell/solide8c/s8cfint_reg.F [hunk 1]
```diff
@@ -0,0 +1,937 @@
...
+     . STI1,STI2,STI3,STI4,STI5,STI6,STI7,STI8
+      my_real, DIMENSION(:) ,ALLOCATABLE   :: 
+     . BTB11,BTB12,BTB13,BTB14,BTB15,BTB16,BTB17,BTB18,
+     . BTB22,BTB23,BTB24,BTB25,BTB26,BTB27,BTB28,BTB33,
+     . BTB34,BTB35,BTB36,BTB37,BTB38,BTB44,BTB45,BTB46,
+     . BTB47,BTB48,BTB55,BTB56,BTB57,BTB58,BTB66,BTB67,
+     . BTB68,BTB77,BTB78,BTB88
+      INTEGER, DIMENSION(:), ALLOCATABLE   ::
+     . POS1,POS2,POS3,POS4,POS5,POS6,POS7,POS8
+      my_real, POINTER, DIMENSION(:) ::
```
Comment (sebastienVilleneuve) [inline engine/source/elements/thickshell/solide8c/s8cfint_reg.F:103]: @vdavaze 

Can you use ALLOCATABLE array in place of array ?
( it is not recommended to use POINTER if we have other possible solution)

Thanks,
Sébastien

**Review by sebastienVilleneuve — CHANGES_REQUESTED** [review]
@vdavaze 

Can you use ALLOCATABLE array in place of array ?
( it is not recommended to use POINTER if we have other possible solution)

Thanks,
Sébastien

File: engine/source/elements/thickshell/solide8c/s8cfint_reg.F (hunk 1 above)
Comment (vdavaze) [reply engine/source/elements/thickshell/solide8c/s8cfint_reg.F]: Hi @sebastienVilleneuve 

The code has been fixed :-) . Pointers have been removed from the code. 

Thanks a lot. 
Valentin.

File: engine/source/elements/thickshell/solide8c/s8cfint_reg.F [hunk 2]
```diff
@@ -0,0 +1,916 @@
...
+C-----------------------------------------------
+C   G l o b a l   P a r a m e t e r s
+C-----------------------------------------------
+#include      "parit_c.inc"
+C-----------------------------------------------
+C   D u m m y   A r g u m e n t s
+C-----------------------------------------------
+      INTEGER, INTENT(IN) :: NFT,NLAY,NPTR,NPTS,IR,IS,
+     .                       NEL,IMAT,ITASK,ICR,ICS,ICT
+      INTEGER, DIMENSION(NEL) :: NC1,NC2,NC3,NC4,NC5,NC6,NC7,NC8
```
Comment (sebastienVilleneuve) [inline engine/source/elements/thickshell/solide8c/s8cfint_reg.F:57]: some INTENT statement missing

File: engine/source/elements/thickshell/solide8c/s8cfint_reg.F [hunk 3]
```diff
@@ -0,0 +1,916 @@
...
+      my_real, DIMENSION(NEL), INTENT(IN) :: 
+     .   OFFG
+      my_real, INTENT(INOUT) ::
+     .   DT2T
+      my_real, DIMENSION(NEL), INTENT(IN) :: 
+     .   VOL,VOL0,AREA
+      TYPE(NLOCAL_STR_) :: NLOC_DMG
```
Comment (sebastienVilleneuve) [inline engine/source/elements/thickshell/solide8c/s8cfint_reg.F:64]: some INTENT statement missing

**Review by sebastienVilleneuve — CHANGES_REQUESTED** [review]
@vdavaze 

having a quick look to the code , I've found some minor issue ( INTENT missing)

Sébastien

File: engine/source/elements/thickshell/solide8c/s8cfint_reg.F (hunk 3 above)
Comment (vdavaze) [reply engine/source/elements/thickshell/solide8c/s8cfint_reg.F]: Missing intent have been added 👍🏻

Comment (sebastienVilleneuve) [reply engine/source/elements/thickshell/solide8c/s8cfint_reg.F]: Thanks !!

**Review by sebastienVilleneuve — APPROVED** [review]

**Pull request merged**